Doctrine/Orm/AutoRouteListener.php

```diff
@@ -141,16 +154,41 @@ public function postFlush(PostFlushEventArgs $args)
     public function postPersist(LifecycleEventArgs $eventArgs)
     {
         $entity = $eventArgs->getEntity();
-        if ($this->isAutoRouteable($entity)) {
+        if ($entity = $this->tryGetAutoRouteable($entity)) {
             $manager = $eventArgs->getEntityManager();
             $unitOfWork = $manager->getUnitOfWork();
             /** @var AutoRoute $autoRoute */
             foreach ($entity->getRoutes() as $autoRoute) {
                 if (!$autoRoute->getContentId()) {
-                    $autoRoute->setContentId($this->getEntityMetadata($manager, $entity)->getIdentifierValues($entity));
+                    $id = $this->getEntityMetadata($manager, $entity)->getIdentifierValues($entity);
+                    $autoRoute->setContentId($id);
+                    $this->rewriteIdOnNameField($autoRoute, $id, false);
```

Comment:
I don't like so much this naming convention. The last boolean is not much clear about the intention of the rewrite method, can you refactor it to be more clear ?